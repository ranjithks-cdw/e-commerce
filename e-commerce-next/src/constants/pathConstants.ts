const PATH_CONSTANTS = {
    HOME: {
        PATH: '/',
        TITLE: 'Home',
    },
    PRODUCTS: {
        PATH: '/products',
        TITLE: 'Products',
    },
    ORDERS: {
        PATH: '/orders',
        TITLE: 'Orders',
    },
    CART: {
        PATH: '/cart',
        TITLE: 'Cart',
    },
    PROFILE: {
        PATH: '/profile',
        TITLE: 'Profile',
    },
    LOGIN: {
        PATH: '/login',
        TITLE: 'Login',
    }
};

export default PATH_CONSTANTS;