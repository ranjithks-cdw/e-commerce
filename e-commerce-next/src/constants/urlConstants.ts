export const URLS = {
    FEATURES: process.env.FEATURES,
}