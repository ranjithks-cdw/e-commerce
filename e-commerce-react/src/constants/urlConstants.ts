export const URLS = {
    FEATURES: process.env.REACT_APP_FEATURES,
};