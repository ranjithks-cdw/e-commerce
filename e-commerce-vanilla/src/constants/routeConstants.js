export const APP_PATHS = {
    HOME: {
        TITLE: 'Home',
        PATH: '/',
    },
    PRODUCTS: {
        TITLE: 'Products',
        PATH: '/products',
    },
    PRODUCT_DESCRIPTION: {
        TITLE: 'Product Description',
        PATH: '/products/:productId',
    },
    CART: {
        TITLE: 'Cart',
        PATH: '/cart',
    },
    ORDERS: {
        TITLE: 'Orders',
        PATH: '/orders',
    },
    PROFILE: {
        TITLE: 'Profile',
        PATH: '/profile',
    },
    LOGIN: {
        TITLE: 'Login',
        PATH: '/login',
    },
    LOGOUT: {
        TITLE: 'Logout',
        PATH: '/logout',
    },
};
