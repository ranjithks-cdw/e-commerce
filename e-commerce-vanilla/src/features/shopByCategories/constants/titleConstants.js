export const SBCFeatureTitles = {
    PRODUCT_CATEGORIES: 'Product Categories',
    VIEW_ALL: 'View All',
};