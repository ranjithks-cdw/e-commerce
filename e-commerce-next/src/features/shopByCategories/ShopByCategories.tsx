import Link from "next/link";
import { SBCFeatureTitles } from "./constants/titleConstants";
import CategoryCard from "./components/categoryCard/categoryCard";
import './styles.css';
import PATH_CONSTANTS from "@/constants/pathConstants";

const ShopByCategories = async () => {
    const shopByCategories = await getCategoriesData();

    const cardsContent = shopByCategories?.map(content => <CategoryCard key={content.categoryName} {...content} />)

    return ( 
        <div className="shop-by-categories">
            <p className="feature-title">{SBCFeatureTitles.PRODUCT_CATEGORIES} <span className="navigate-link"><Link href={PATH_CONSTANTS.PRODUCTS.PATH}>{SBCFeatureTitles.VIEW_ALL}</Link></span></p>
            <div className="category-content">
                {cardsContent}
            </div>
        </div>
     );
};

const getCategoriesData = async () => {
    const shopByCategoriesData = [
        {
            categoryName: "Title 1",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 2",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 3",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 4",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 5",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
      ];
    
      return shopByCategoriesData;
}
 
export default ShopByCategories;