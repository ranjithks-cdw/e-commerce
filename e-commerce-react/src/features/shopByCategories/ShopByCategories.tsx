import { Link } from "react-router-dom";
import { SBCFeatureTitles } from "./constants/titleConstants";
import APP_PATHS from "../../constants/routeConstants";
import React, { useEffect, useState } from "react";
import { sbcTypes } from "./types/categoryTypes";
import CategoryCard from "./components/categoryCard/CategoryCard";
import './styles.css';

const ShopByCategories = () => {
    const [categoryData, setCategoryData] = useState<sbcTypes[]>([]);

    useEffect(() => {
        const data = [
            {
                categoryName: "Title 1",
                categoryImageUrl: "https://placehold.co/600x400",
            },
            {
                categoryName: "Title 2",
                categoryImageUrl: "https://placehold.co/600x400",
            },
            {
                categoryName: "Title 3",
                categoryImageUrl: "https://placehold.co/600x400",
            },
            {
                categoryName: "Title 4",
                categoryImageUrl: "https://placehold.co/600x400",
            },
            {
                categoryName: "Title 5",
                categoryImageUrl: "https://placehold.co/600x400",
            },
        ];

        setCategoryData(data);
    }, []);

    const cardsContent = categoryData?.map(data => <CategoryCard key={data.categoryName} {...data} />)

    return ( 
        <div className="shop-by-categories">
            <p className="feature-title">{SBCFeatureTitles.PRODUCT_CATEGORIES} <span className="navigate-link"><Link to={APP_PATHS.PRODUCTS.PATH}>{SBCFeatureTitles.VIEW_ALL}</Link></span></p>
            <div className="category-content">
                {cardsContent}
            </div>
        </div>
     );
};
 
export default ShopByCategories;