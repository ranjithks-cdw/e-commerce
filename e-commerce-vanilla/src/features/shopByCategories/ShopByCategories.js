import { APP_PATHS } from "../../constants/routeConstants.js";
import { createElement } from "../../utils/htmlUtils.js";
import CardContent from "./components/cardContent/CardContent.js";
import { SBCFeatureTitles } from "./constants/titleConstants.js";

const ShopByCategories = {
    addComponent: () => addComponent(),
    removeComponent: () => removeComponent(),
};

const createDOM = async () => {
    const shopByCategories = createElement('div', {
        className: 'shop-by-categories',
    });

    const categoryContent = createElement('div', {
        className: 'category-content',
    });

    const featureTitle = createElement('p', {
        className: 'feature-title',
        textContent: SBCFeatureTitles.PRODUCT_CATEGORIES
    });

    const viewAll = createElement('span', {
        className: 'navigate-link',
    });
    const link = createElement('a', {
        href: APP_PATHS.PRODUCTS.PATH,
        textContent: SBCFeatureTitles.VIEW_ALL
    });
    viewAll.appendChild(link);

    const contentWrapper = new DocumentFragment();
    const data = await getCategoriesData();
    data.forEach(content => {
        const cardContent = CardContent.create({...content});
        contentWrapper.appendChild(cardContent);
    });

    
    featureTitle.appendChild(viewAll);
    shopByCategories.appendChild(featureTitle);
    categoryContent.appendChild(contentWrapper);
    shopByCategories.appendChild(categoryContent);
    return shopByCategories;
};

const addComponent = () => {
    const component = createDOM();
    return component;
};

const removeComponent = () => {

};

const getCategoriesData = async () => {
    const shopByCategoriesData = [
        {
            categoryName: "Title 1",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 2",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 3",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 4",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
        {
            categoryName: "Title 5",
            categoryImageUrl: "https://dummyimage.com/600x400/aaaaaa/f2f2f2.jpg",
        },
      ];
    
      return shopByCategoriesData;
};

export default ShopByCategories;