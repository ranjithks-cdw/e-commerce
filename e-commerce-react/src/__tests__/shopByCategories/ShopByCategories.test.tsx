import { render, screen, waitFor } from "@testing-library/react";
import renderer from 'react-test-renderer';
import ShopByCategories from "../../features/shopByCategories/ShopByCategories";
import { SBCFeatureTitles } from "../../features/shopByCategories/constants/titleConstants";
import { MemoryRouter } from "react-router";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // Use the actual implementation for other functions
    Link: jest.fn().mockImplementation(({ to, children }) => <a href={to}>{children}</a>),
}));

describe('ShopByCategories', () => {
    it("renders component", async () => {
        render(
            <MemoryRouter>
                <ShopByCategories />
            </MemoryRouter>
        );
        const titleText = await waitFor(() => screen.findByText(SBCFeatureTitles.PRODUCT_CATEGORIES), {timeout: 2000});
        expect(titleText).toBeInTheDocument();
    })

    it("matches snapshot", () => {
        const container = renderer.create(
            <MemoryRouter>
                <ShopByCategories />
            </MemoryRouter>
        ).toJSON;
        expect(container).toMatchSnapshot();
    })

    jest.restoreAllMocks();
})