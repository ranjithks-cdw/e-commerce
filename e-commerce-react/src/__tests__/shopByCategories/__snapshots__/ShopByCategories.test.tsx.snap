// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopByCategories matches snapshot 1`] = `[Function]`;
